core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -125,8 +130,8 @@ protected void inboundHeadersReceived(Metadata.Headers headers) {
    *
    * @param frame the received data frame. Its ownership is transferred to this method.
    */
-  protected void inboundDataReceived(ReadableBuffer frame) {
-    Preconditions.checkNotNull(frame, "frame");
```

Comment:
Redundant.  A stack trace will point to exactly what line caused the problem.  Additionally, the string frame is a minor maintainability drag, since automated renames of this variable will not update the name.  While this project is small, it doesn't help much, but as it grows it helps.

If there was a more meaningful message here, something along the lines of "Invalid FrameBuffer", it would warrant staying.  In general, less code is better since it speeds up reading.  This is also why static imports are helpful, since the word "Preconditions" doesn't add much meaning to the code.